lib/matplotlib/tests/test_category.py

```diff
@@ -187,6 +187,65 @@ def test_plot_1d_missing(self):
         self.axis_test(ax.yaxis, self.dmticks, self.dmlabels, self.dmunit_data)
 
     @cleanup
+    @pytest.mark.usefixtures("data")
+    def test_plot_bytes(self):
+        counts = np.array([4, 6, 5, 1])
+        fig, ax = plt.subplots(ncols=3)
+
+        ax[0].bar(self.d, counts)
+
+        types = [v.encode('ascii') for v in self.d]
+        ax[1].bar(types, counts)
+
+        types = np.array(types)
+        ax[2].bar(types, counts)
+
+        fig.canvas.draw()
+
+        # All three plots should look like the string one.
+        self.axis_test(ax[1].xaxis,
+                       ax[0].xaxis.get_majorticklocs(),
+                       lt(ax[0].xaxis.get_majorticklabels()),
+                       ax[0].xaxis.unit_data)
+        self.axis_test(ax[2].xaxis,
+                       ax[0].xaxis.get_majorticklocs(),
+                       lt(ax[0].xaxis.get_majorticklabels()),
+                       ax[0].xaxis.unit_data)
+
+    @cleanup
+    def test_plot_numlike(self):
```

Comment:
hmm, so the way to do that if you don't want to write the fixed values (though the expected values is probably better testing practice*) is probably to just always create an axis 0 and then parameterize out axis 1 and axis 2. 
* ticklocs is the fixed assignment, in this case with 3 values ['a', 'c', 'c', 'd'] becomes ['a;, 'c', 'd'] since they're analogous to x values, the ticklocs are [0, 1, 2] and the labels are ['a', 'c', 'd'], which also creates a mismatch between the x and counts and so ['a', 'b', 'c', 'd'] should probably be used explicitely.